apps/discovery_api/lib/discovery_api/data/visualization_migrator.ex

```diff
@@ -0,0 +1,56 @@
+defmodule DiscoveryApi.Data.VisualizationMigrator do
+  @moduledoc """
+  Module to ensure all derived fields of visualizations are up to date.
+  """
+  alias DiscoveryApi.Schemas.Visualizations
+  alias DiscoveryApi.Schemas.Visualizations.Visualization
+
+  use GenServer, restart: :transient
+
+  import Ecto.Query, only: [from: 2]
+
+  require Logger
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, nil)
+  end
+
+  def init(_) do
+    {:ok, nil, {:continue, :run}}
+  end
+
+  def handle_continue(:stop, _) do
+    {:stop, :normal, nil}
+  end
+
+  def handle_continue(:run, _) do
+    case migrate() do
+      :ok ->
+        Logger.info("Visualization Migration Successful")
+        {:noreply, nil, {:continue, :stop}}
+
+      :error ->
+        Logger.info("Visualization Migration Failed. Retrying.")
+        Process.sleep(2000)
+        {:noreply, nil, {:continue, :run}}
+    end
+  end
+
+  defp migrate() do
+    try do
+      Logger.info("Starting Migration of Visualizations")
+      query = from(v in Visualization, where: is_nil(v.valid_query))
```

Comment:
I'd love it if the relationship between valid_query and the stuff done in `Visulizations` was more clear, but I can't think of a good way off the top of my head.

Reply:
Maybe the query for non-updated visualizations could be owned by `Visualizations` ? That way, the field it's checking doesn't need to be spread across two modules?